pybossa/api/__init__.py

```diff
@@ -417,6 +420,92 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def get_user_preferences(user_name=None):
+    """API endpoint for loading account user preferences.
+    Returns a JSON object containing the user account preferences.
+    """
+    if not user_name:
+        return abort(404)
+
+    user = user_repo.get_by_name(user_name)
+    if not user:
+        return abort(404)
+
+    try:
+        (can_update, disabled_fields, hidden_fields) = can_update_user_info(current_user, user)
+    except Exception:
+        return abort(404)
+
+    if not can_update:
+        return abort(403)
+
+    user_preferences = get_user_pref_metadata(user_name)
+
+    if not user_preferences and user_preferences != {}:
+        return abort(500)
```

Comment:
can the api not sent empty response when no user pref set? if so, these lines can be omitted

Reply:
This provides for an unexpected error. I believe there should always be a response or `{}` for an empty object. If the result is `None`, there may be a problem with the record.

Reply:
Confirmed an empty response would return a value. Removed check via https://github.com/bloomberg/pybossa/pull/675/commits/bee20e91c6c97a4361c7d8149844cbc7c0148565.

```bash
>>> json.dumps(None)
'null'
```